systemtest/benchtest/main.go

```diff
@@ -240,3 +236,33 @@ func Run(allBenchmarks ...BenchmarkFunc) error {
 	}
 	return nil
 }
+
+func warmup(agents int, events uint) error {
+	// Assume a base ingest rate of at least 5000 per second, and dynamically
+	// set the context timeout based on this ingest rate, or if lower, default
+	// to 15 seconds. The default 5000 / 5000 = 1, so the default 15 seconds
+	// will be used instead.
+	timeout := time.Duration(math.Max(float64(events/5000), 15)) * time.Second
+	var wg sync.WaitGroup
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	for i := 0; i < agents; i++ {
+		h, err := newEventHandler(`*.ndjson`, getNewLimiter())
+		if err != nil {
+			return fmt.Errorf("unable to create warm-up handler: %w", err)
+		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			h.WarmUpServer(ctx, events)
```

Comment:
We should handle errors from the `WarmUpServer` function (maybe by using a return channel)